src/AbstractAppliance.js

```diff
@@ -0,0 +1,119 @@
+import assert from 'assert'
+import EventEmitter from 'events'
+import {
+	AbstractInstantiationError,
+	NotImplementedError,
+} from './errors'
+import PayloadBuffer from './PayloadBuffer'
+import { isPayloadInstance } from './utils/payload'
+
+/**
+ * The Abstract Appliance defines the shape of an Appliance and implements
+ * a few universal pieces of functionality, such as:
+ *
+ * - Event emission
+ * - Payload validation process flow
+ * - Appliance setting resolution
+ *
+ * AbstractAppliance methods prefixed with `_` require implementation.
+ * Other methods, aside from the constructor, should not be overridden.
+ *
+ * For more information about the TV Kitchen architecture visit:
+ * https://github.com/tvkitchen/tv-kitchen/blob/master/docs/ARCHITECTURE.md
+ */
+class AbstractAppliance {
+	settings = {}
+
+	emitter = new EventEmitter()
+
+	payloadBuffer = new PayloadBuffer()
+
+	constructor(overrideSettings = {}) {
+		if (this.constructor === AbstractAppliance) {
+			throw new AbstractInstantiationError('AbstractAppliance')
+		}
+		Object.assign(
+			this.settings,
+			overrideSettings,
+		)
+	}
+
+	/**
+	 *******************************************
+	 ************ ABSTRACT METHODS *************
+	 *******************************************
+	 */
+
+	/**
+	 * Getter for the list of data types accepted by the appliance.
+	 *
+	 * OVERRIDE WHEN EXTENDING
+	 *
+	 * @return {Array[String]} The list of data types accepted by the appliance.
+	 */
+	getInputTypes = () => {
+		throw new NotImplementedError('getInputTypes')
+	}
+
+	/**
+	 * Getter for the list of data types produced by the appliance.
+	 *
+	 * OVERRIDE WHEN EXTENDING
+	 *
+	 * @return {Array[String]} The list of data types produced by the appliance.
+	 */
+	getOutputTypes = () => {
+		throw new NotImplementedError('getOutputTypes')
+	}
+
+	/**
+	 * Asserts that a given payload is actually a payload.
+   *
+   * OVERRIDE WHEN EXTENDING
+   *
+	 * @param  {Payload} payload The payload that we want to validate.
+	 * @return {Boolean}         The result of the validation check.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	isValidPayload = async (payload) => {
+		throw new NotImplementedError('isValidPayload')
+	}
+
+	/**
+	 * Invokes the appliance on any unprocessed data in the appliance buffer.
+	 *
+	 * @return {Boolean}         Whether or not the appliance had enough data to work with.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	invoke = async () => {
+		throw new NotImplementedError('process')
```

Comment:
Looks like you renamed the method but forgot to change this. Should be `'invoke'`.